app/scripts/modules/openstack/common/selectField.directive.js

```diff
@@ -48,8 +48,10 @@ module.exports = angular.module('spinnaker.openstack.selectField.directive', [
 
             //if the selection changed as a result of updating the options
             if (scope.model !== previousSelection) {
-              //ensure that the change handler is invoked
-              scope.onChange();
+              //ensure that the change handler is invoked... but give time for the two-way binding to propagate the value
+              if( scope.onChange ){
+                $timeout(scope.onChange);
+              }
```

Comment:
Yes, that is possible but it is a little bit less efficient as it results in a noop function callback being queued and a deferred/promise to be allocated unnecessarily.